src/store/reducers/agentsReducer.js

```diff
@@ -80,6 +85,27 @@ export function agentsReducer(state = initialState, action) {
       };
     case DELETE_AGENT_FAIL:
       return { ...state, status: 'fail', error: action.payload.error };
+    case TOGGLE_FAVORITE_START:
+      return {
+        ...state,
+        addAgentPopupVisibility: true,
+      };
+    case TOGGLE_FAVORITE_SUCCESS:
+      return {
+        ...state,
+        toggleFavoriteAgentInprogress: false,
+        items: state.items.map((item) => {
+          if (item.id !== action.payload.agentId) {
+            item.isFavorite = !item.isFavorite;
+          }
+          return item;
+        }),
+      };
```

Comment:
the latest agent already have the updated isFavorite property, you do not need to handle the logic here.